webrender/src/resource_cache.rs

```diff
@@ -588,7 +616,15 @@ impl ResourceCache {
             tile,
         };
         let image_info = &self.cached_images.get(&key);
-        self.texture_cache.get(&image_info.texture_cache_handle)
+
+        // If an image is not in cached_images, just return an invalid CacheItem
+        // for that request. That invalid item will be skipped during the rendering.
+        //
+        // TODO(Jerry): add a debug option to fill the corresponding area for
+        // that invalid CacheItem.
+        image_info.map_or(CacheItem::invalid(), |image_info| {
```

Comment:
we call `get_glyphs` when populating batches. Wouldn't it be simpler to just return an error here, so that a corresponding instance can be discarded?

Reply:
We call get_glyphs at [1]. If there is any error inside get_glyphs()[2], the lambda expression at [3] will not be called(which will push the instance to draw list).

[1]
https://github.com/servo/webrender/blob/e11297a240b29bd57154d37ecb402d0b9aa299e6/webrender/src/tiling.rs#L513
[2]
https://github.com/servo/webrender/pull/1564/files#diff-77cbdf7ba9ebae81feb38a64c21b8454R630
[3]
https://github.com/servo/webrender/blob/e11297a240b29bd57154d37ecb402d0b9aa299e6/webrender/src/tiling.rs#L518